### ISSUE TITLE
`DenseNetEfficientMult` not giving same forwarding result as `DenseNetEfficient`

### ISSUE THREAD

**Issue by ZhengRui** [gpleiss/efficient_densenet_pytorch#17]
I tried to train on multi gpus, and after a lot of tries I found `DenseNetEfficientMulti` is not giving same output as `DenseNetEfficient`, actually `DenseNetEfficientMulti`'s output depends on how many `device_ids` specified in `DataParallel`. However, when only specify one gpu, it indeed gives same result as `DenseNetEfficient`. And when selected gpus are fixed, `DenseNetEfficientMulti`'s outputs are also fixed.

If i just use `DenseNetEfficient` for multi gpu case, it will say `Tensors are on different gpus`. I guess there are some buffer which caused this issue.

~@taineleau @gpleiss  would you either make `DenseNetEfficient` being able to support multi gpu case, or fix the bug of `DenseNetEfficientMulti`, I think `DenseNetEfficientMulti` might has some bugs as its output depends on the number of gpus used. Any insights to solve this will be helpful, thanks.~

**Comment by ZhengRui** [comment]
I found the issue is because for me to use densenet121 structure, I have to change the first few layers a little bit, especially the max pooling layer might cause this issue. Will confirm this soon.

**Comment by ZhengRui** [comment]
I found it is related to the batchnorm layers. However, I also tested the batchnorm layer inside the denseblock, it also gives different results when using single gpu or multi gpus. Just to confirm, @taineleau is the batchnorm layer inside each dense layer asynchronized?

**Comment by ZhengRui** [comment]
Ok, things are clear to me now, in eval() mode, the multi gpu and single gpu will give same results. The different results I got even in eval() mode is because my feature extraction script doesn't take into account of the multi gpu case. In train() mode, it is expected multi gpu and single gpu will give different results due to batch norm layer. I am closing this issue now.
